NitroxClient/GameLogic/InitialSync/PlayerPositionInitialSyncProcessor.cs

```diff
@@ -44,33 +48,53 @@ public override IEnumerator Process(InitialPlayerSync packet, WaitScreen.ManualW
             Optional<NitroxId> subRootId = packet.PlayerSubRootId;
             if (!subRootId.HasValue)
             {
+                yield return WaitForWorldLoad();
                 yield break;
             }
 
             Optional<GameObject> sub = NitroxEntity.GetObjectFrom(subRootId.Value);
             if (!sub.HasValue)
             {
                 Log.Error("Could not spawn player into subroot with id: " + subRootId.Value);
+                yield return WaitForWorldLoad();
                 yield break;
             }
 
             if (!sub.Value.TryGetComponent(out SubRoot subRoot))
             {
                 Log.Debug("SubRootId-GameObject has no SubRoot component, so it's assumed to be the EscapePod");
+                yield return WaitForWorldLoad();
                 yield break;
             }
 
             // If player is not swimming
             Player.main.SetCurrentSub(subRoot);
             if (subRoot.isBase)
             {
+                // If the player's in a base, we don't need to wait for the world to load
                 yield break;
             }
             Transform rootTransform = subRoot.transform;
             Quaternion vehicleAngle = rootTransform.rotation;
             position = vehicleAngle * position;
             position = position + rootTransform.position;
             Player.main.SetPosition(position);
+            yield return WaitForWorldLoad();
+        }
+
+        public static IEnumerator WaitForWorldLoad()
```

Comment:
Maybe we should extract this Enumerator to a proper class, so PlayerTeleportedProcessor and maybe future class are not coupled with that one